송현명 todo 과제제출

mung96 opened the pull request.

mung96 committed:
setting: 초기세팅

-css초기세팅
-palette에 주요 색 모음
-컴포넌트 구성 (템플릿,컨테이너,아이템)

mung96 committed:
feat: 템플릿역할(TodoTemplate.js) ui 구현

mung96 committed:
feat: work아이템을 담는 컨테이너 ui구현

mung96 committed:
feat: work에 해당하는 아이템 UI 구현

mung96 committed:
feat: work에 title,body 동적으로 할당

mung96 committed:
feat: 새로운 item 작성 기능

- +버튼 클릭시 회색 작성 칸 등장
-취소 버튼 클릭시 삭제

mung96 committed:
feat: container에 newItem 생성 기능 구현

-reload시 마지막 아이템이 사라지는 버그 있음

mung96 committed:
fix: jsonLocalStorage에 한단계씩 늦게 setItem되는 버그 수정

mung96 committed:
feat: item생성중 x표 누르면 jsonLocalStorage에서 삭제

-but 실제 ui에서 삭제가 안됨

mung96 committed:
fix:  item ui 삭제안되는 버그 수정

mung96 committed:
Merge pull request #1 from mung96/feature/remove

item 삭제 기능 구현

mung96 committed:
feat: 수정 버튼 클릭시 ui 구현

mung96 committed:
feat: item수정 기능 구현

mung96 committed:
Merge pull request #2 from mung96/feature/edit

item 수정 기능 구현

mung96 committed:
feat: move 버튼 ui 구현

-move 누르면 localstorage에 todos에 null이 쌓임

mung96 committed:
bug

mung96 committed:
feat: move버튼 클릭시 localstorage 변화 구현

mung96 committed:
refactor: 변수명 변경

mung96 committed:
Merge pull request #3 from mung96/feature/move

-item 1개 생성 이후 localstorage에서는 이동을 함.
-item 1개 이동후 다음 item생성시부터 localstorage 에러

mung96 committed:
refactor: 컨테이너 제목 수정

mung96 committed:
docs:README.md 작성

mung96 committed:
Merge pull request #4 from mung96/readme

Readme

mung96 committed:
Merge pull request #5 from mung96/readme

Readme

s0ojin commented on src/components/EditItemForm.js:
```diff
@@ -0,0 +1,41 @@
+  return (
+    <form className="EditItemForm" onSubmit={handleSubmitBtn}>
+      <div className="colors">
+        <div></div>
```
이 태그는 왜 있는건가요?

s0ojin commented on src/lib/palette.js:
palette이거 어디서쓰이고 있나요?

s0ojin commented on src/components/TodoTemplate.js:
```diff
@@ -0,0 +1,48 @@
+    <div className="TodoTemplate">
+      <h1 className="title">RoadMap</h1>
+      <div className="content">
+        <WorkContainer type="todos" handleMoveBtnClick={handleMoveBtnClick}>
```
map을 돌려볼까요??

s0ojin commented on src/components/TodoTemplate.js:
```diff
@@ -0,0 +1,48 @@
+    <div className="TodoTemplate">
+      <h1 className="title">RoadMap</h1>
+      <div className="content">
+        <WorkContainer type="todos" handleMoveBtnClick={handleMoveBtnClick}>
```
work라는 네이밍을 봤을 때 바로 일정이라고 짐작하기 어렵운 것 같습니다!

s0ojin replied:
같은 맥락에서 왜 부모는 투두템플릿인데 일정은 워크콘테이너인가요? 그래서 더 헷갈리는 것 같습니다!

s0ojin commented on src/components/TodoTemplate.js:
```diff
@@ -0,0 +1,48 @@
+import { useState } from "react";
+
+const TodoTemplate = ({ children }) => {
+  const [startItems, setStartItems] = useState([]);
```
startItems와 endItems가 뭔가요?

s0ojin commented on src/components/WorkContainer.js:
```diff
@@ -0,0 +1,104 @@
+            handleMoveBtnClick={handleMoveBtnClick}
+            type={type}
+            handleLocalStorageChange={handleLocalStorageChange}
+          ></WorkItem>
```
빈태그 일때는 셀프 클로징 태그를 사용해주시면 보다 깔끔한 코드가 될 것 같습니다!

s0ojin reviewed (commented):
고생하셨습니다! 쭉보면서 눈에 보이는 부분만 리뷰를 달았는데 못다한 부분은 대면으로 더 이야기 해보시죠!

mung96 committed:
Merge pull request #6 from mung96/develope

Todo 1주차 PR

mung96 committed:
refactor: 필요없는 파일 삭제, js파일 jsx로 변경

mung96 committed:
refactor: 변수명, 클래스명 변경

-WorkItem => Card
-WorkContainer => Board
-TodoTemplate => Template
-EditItemForm=>EditCardForm
-NewItemForm=>NewCardForm
-변수명 item은 모두 card로

mung96 committed:
refactor: onUpdate 함수  리팩터링

-for문을  filter와 concat 으로 변경

mung96 committed:
refactor: 삭제버튼 코드 리팩터링(handleCloseBtnClick)

-for문을 filter 변경

mung96 committed:
refactor: Board 컴포넌트 생성 과정 리팩터링

-map을 이용

mung96 committed:
refactor: 생성, 수정 폼 부분 리팩터링

-mode변수 없이 리팩터링

mung96 committed:
refactor: icon폴더명 asset으로 변경

-확장성 고려

mung96 committed:
refactor: 수정버튼과 닫기버튼 ui 수정

-버튼 위치 교환

mung96 committed:
Merge pull request #7 from mung96/refactor

피드백 반영 refactor (아직 버그 수정X)

mung96 committed:
refactor: id를 board가 공유하도록 설정

-useRef 사용

mung96 committed:
refactor: handleCloseBtnClick을 Card의 함수로 이동

-의도: 카드에 추가적인 버튼이 생길 때 board에서 계속 함수를 만들어 Card에게 넘기는 방식이 비효율적이라 판단

mung96 committed:
refactor: Card 수정 기능을 Card의 함수로 이동

mung96 committed:
Merge pull request #8 from mung96/remove-edit

카드 수정,삭제 관련 refactor (함수 권한 변경)

mung96 committed:
refactor: 모든 Board의 카드 배열을 Template에서 관리

mung96 committed:
refator: 수정, 삭제 기능에서 Boards 변경가능

mung96 committed:
feat: move버튼 클릭시 Board 이동 구현

mung96 committed:
refactor: 카드 수정 UI 파일 분리

mung96 committed:
Merge pull request #9 from mung96/feature/move

Card 이동 다른 Board 기능 구현

mung96 committed:
feat: 색을 입힐 수 있는 버튼 UI 구현

mung96 committed:
refactor: 카드 생성시 등록과 취소 버튼 함수 NewCardForm에서 정의

mung96 committed:
feat: Card 생성시 색상 설정 기능 구현

mung96 committed:
feat: 카드 수정시 색 변경 기능 구현

mung96 committed:
Merge pull request #10 from mung96/card-color

카드 색상 선택 기능 구현

mung96 committed:
fix: 카드 수정시 UI 크기 조절

mung96 committed:
fix: 카드가 항상 id 순으로 정렬되도록 수정

mung96 committed:
fix: card 제목 길어지면 card 밖으로 나가는 버그 수정

mung96 committed:
fix: 이동버튼 hover시에만 표시

mung96 committed:
fix: 추가(+)버튼 ui 수정

mung96 committed:
fix: board 3개가 독립적으로 움직이도록 변경

mung96 committed:
fix: 카드 수정시 기본값을 원래 카드 색상으로 선택

mung96 committed:
fix: 리로드시 이동 버그 수정

mung96 committed:
Merge pull request #11 from mung96/fix

버그수정

mung96 committed:
feat: darkMode 구현

mung96 committed:
feat: 카드 생성시 색상 focus 구현

mung96 committed:
Merge pull request #12 from mung96/darkmode

다크모드 구현
